docs/csharp/pattern-matching.md

```diff
@@ -2,31 +2,192 @@
 title: Pattern Matching | C# Guide
 description: Learn about pattern matching expressions in C#
 keywords: .NET, .NET Core, C#
-ms.date: 10/04/2016
+ms.date: 12/09/2016
+ms.author: wiwagn
 ms.topic: article
 ms.prod: .net
 ms.technology: devlang-csharp
 ms.devlang: csharp
 ms.assetid: 1e575c32-2e2b-4425-9dca-7d118f3ed15b
 ---
 
-# 🔧 Pattern Matching
+# Pattern Matching #
 
-> **Note**
-> 
-> This topic hasn’t been written yet! 
->
-> We welcome your input to help shape the scope and approach. You can track the status and provide input on this
-> [issue](https://github.com/dotnet/docs/issues/1114) at GitHub.
-> 
-> If you would like to review early drafts and outlines of this topic, please leave a note with your contact information in the issue.
->
-> Learn more about how you can contribute on [GitHub](https://github.com/dotnet/docs/blob/master/CONTRIBUTING.md).
->
+Pattern matching provides a control flow mechanism that is much more
+generalized than is possible with traditional inheritance hierarchies
+available in Object Oriented practices.
 
-<!--
+You are probably familiar with building class hierarchies and creating
+virtual methods and overriden methods to customize object behavior based
+on the runtime type of the object.
 
-<< Build the sample in an additive way >>
-<< Do the shape thing for the https://docs.microsoft.com/en-us/dotnet/articles/fsharp/language-reference/discriminated-unions shapes >>
+Those techniques aren't possible for data that isn't structured in a class
+hierarchy. In those cases, you'd need to utilize traditional procedural
+imperative constructs like `if` or `switch` statements to manage the control
+flow.
 
--->
\ No newline at end of file
+The new *pattern matching* expressions enable cleaner syntax to examine data
```

Comment:
Hmm. I was going from the spec notes here: https://github.com/dotnet/roslyn/blob/features/patterns/docs/features/patterns.md#match-expression  

I'll use 'constructs' here, and read the rest of the doc for the use of both expressions and statements.